src/io/pithos/operations.clj

```diff
@@ -125,13 +125,26 @@
     ([nickname val]
      (parse-int nickname nil val)))
 
+(defn check-range
+  [max [has-range? start end]]
+  (let [e (ex-info "unsatisfiable range"
+                   {:type :invalid-argument
+                    :arg "range"
+                    :val (format "start=%s,end=%s" start end)
+                    :status-code 416})]
+    (cond
+      (> start end) (throw e)
+      :else         [has-range? start end])))
```

Comment:
here i'm only checking that the provided range makes sense. if it does I return what I was given.
The destructuring trick I could have used:

```clojure
(defn check-range
   [max [_ start end :as range]]
     ....
        :else range)
```